sklearn/feature_selection/from_model.py

```diff
@@ -201,7 +222,15 @@ def _get_support_mask(self):
         scores = _get_feature_importances(estimator)
         self.threshold_ = _calculate_threshold(estimator, scores,
                                                self.threshold)
-        return scores >= self.threshold_
+        mask = np.zeros_like(scores, dtype=bool)
+        if self.max_features == 'all':
+            self.max_features = scores.size
+        candidate_indices = np.argsort(-scores,
+                                       kind='mergesort')[:self.max_features]
+        mask[candidate_indices] = True
+        if self.threshold is not None:
```

Comment:
If this condition is removed, we need to use ``threshold=-np.inf`` to pass most of the test case. Is that Ok?
However, I've edited the ``max_features`` doc. You can check it below.

> max_features : int, between 0 and number of features, optional.
>         Use ``threshold=-np.inf`` to ignore ``threshold`` and apply
>         ``max_features`` only. `threshold=None` to select at most this many 
>         features that score above the threshold. Any ``int`` between 0 and 
>         number of features will return the desired number of features.